### PR TITLE
Adapt to Wasm 0.2 library

### LINKED ISSUE #3363: use `ocamlPackages.wasm` v2.0.0
https://github.com/NixOS/nixpkgs/commit/713b7b04045268f155dc27b3a5e42ec3fef09104

I think this will be available once we bump `nixpkgs`. May need OCaml-4.13, though.

### PR DESCRIPTION
**Pull request by ggreif**
DON'T LOOK YET, THIS IS A MESS!
----------

Fixes #3363.

### CODE REVIEW

File: src/wasm-exts/utf8.ml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+type codepoint = int
```
Comment (ggreif) [inline src/wasm-exts/utf8.ml:1]: maybe we can remove the other Utf8 decoder and use this only? Ditto the other `.mli` file.

Comment (crusso) [reply src/wasm-exts/utf8.ml:1]: IIRC correctly, this version isn't tail-recursive and can blow up - is the other one written to be tail-recursive perhaps?

Comment (ggreif) [reply src/wasm-exts/utf8.ml:1]: Turns out, this file is not needed, `encoder.ml` references it as `Wasm.utf8`. But it uses it for (short?) identifiers only.

File: src/wasm-exts/free.ml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+let start (s : start) = funcs (var s.it.sfunc)
+
+let module_ (m : module_) =
+  list type_ m.it.types ++
+  list global m.it.globals ++
+  list table m.it.tables ++
+  list memory m.it.memories ++
+  list func m.it.funcs ++
+    (*opt start m.it.start ++*)
```
Comment (ggreif) [inline src/wasm-exts/free.ml:143]: This gives me a type error. Needs fix.

**Comment by nomeata** [comment]
Can we maybe first bump to the new wasm version, and then ocaml and nixpkgs separately? Or is that too tricky?

**Comment by luc-blaeser** [comment]
Just doing some repo cleanup for better focussing on active work:
Closing PR as it seems inactive for a longer time. Please re-open if it is still relevant and after it has been updated with latest master changes.

**Pull request closed without merging**